yt_dlp/extractor/vevo.py

```diff
@@ -149,7 +149,7 @@ def _real_extract(self, url):
 
                 formats.append({
                     'url': version_url,
-                    'format_id': 'http-%s-%s' % (version, video_version['quality']),
+                    'format_id': f'http-{version}-{video_version.get("quality")}',
```

Comment:
You can get the missing quality from the URL, which is already being analysed quite fully:
```diff
---  old/yt-dlp/yt_dlp/extractor/vevo.py
+++ new/yt-dlp/yt_dlp/extractor/vevo.py
@@ -138,6 +138,7 @@
                     fatal=False))
             else:
                 m = re.search(r'''(?xi)
+                    _(?P<quality>[a-z0-9]+)
                     _(?P<width>[0-9]+)x(?P<height>[0-9]+)
                     _(?P<vcodec>[a-z0-9]+)
                     _(?P<vbr>[0-9]+)
@@ -149,7 +150,7 @@
 
                 formats.append({
                     'url': version_url,
-                    'format_id': 'http-%s-%s' % (version, video_version['quality']),
+                    'format_id': 'http-%s-%s' % (version, video_version.get('quality', m.group('quality'))),
                     'vcodec': m.group('vcodec'),
                     'acodec': m.group('acodec'),
                     'vbr': int(m.group('vbr')),
```

Also the yt-dl extractor has tests, missing in this module: consider pulling them in if they work.

Reply:
Blame on youtube-dl repo suggests me that tests were deleted in either youtube-dlc or yt-dlp... why